propagation/baggage.go

```diff
@@ -29,6 +29,19 @@ func (b Baggage) Inject(ctx context.Context, carrier TextMapCarrier) {
 
 // Extract returns a copy of parent with the baggage from the carrier added.
 func (b Baggage) Extract(parent context.Context, carrier TextMapCarrier) context.Context {
+	multiCarrier, isMultiCarrier := carrier.(MultiTextMapCarrier)
```

Comment:
AFAIK type assertions can introduce an additional heap allocation. Extraction of the baggage is on the hot path.
Can you add some benchmark for this method and can use add `benchstat` results to the PR described so that we can see what is the performance overhead?

Can you also implement the `MultiTextMapCarrier` for `TraceContext` so that we can also use the existing `BenchmarkExtract` to validate the performance overhead for `TraceContext` (which is probably more popular than Baggage)?

Reply:
I made a quick checkout of the PR and the `gopls` with `gc_details` enabled has not shown that there is a heap allocation.